src/Void.Core/CoreMessages.fs

```diff
@@ -43,6 +43,20 @@ type CoreCommand =
     | Yank
     interface CommandMessage
 
+[<RequireQualifiedAccess>]
+type CommandHistoryCommand =
+    | MoveToPreviousCommand
+    | MoveToNextCommand
+    interface CommandMessage
+
+
+[<RequireQualifiedAccess>]
+type CommandHistoryEvent =
+    | MovedToCommand of string
+    | MovedToEmptyCommand

```

Comment:
I like that you didn't do `MovedToCommand ""`. That would seem like a big temptation but the semantics of events are very important, and it is a different occurrence when you move out of the history list then when you are moving around in it.